### PR TITLE
Fix bug: can't output Diagnostic message

### PR DESCRIPTION
**Pull request by qinezh**
@vwxyzh @herohua @superyyrrzz @ansyral @hellosnow @chenkennt

### CODE REVIEW

**Comment by ansyral** [comment]
:+1:

File: src/Microsoft.DocAsCode.Common/Loggers/Logger.cs [hunk 1]
```diff
@@ -20,6 +37,7 @@ public static void RegisterListener(ILoggerListener listener)
                 throw new ArgumentNullException(nameof(listener));
             }
 
+            listener.LogLevelThreshold = _logLevelThreshold;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/Logger.cs]: >listener.LogLevelThreshold = _logLevelThreshold; [](start = 12, length = 48)

don't change listener's LogLevelThreshold

**Comment by vwxyzh** [comment]
>                 listener.WriteLine(item);

filter item

---
Refers to: src/Microsoft.DocAsCode.Common/Loggers/ReplayLogListener.cs:139 in 4f47fa5. [](commit_id = 4f47fa5b1164d83f842920872ffecbbb4d57170d, deletion_comment = False)

File: src/Microsoft.DocAsCode.Common/Loggers/ConsoleLogListener.cs [hunk 2]
```diff
@@ -64,6 +64,9 @@ private void ChangeConsoleColor(LogLevel level)
         {
             switch (level)
             {
+                case LogLevel.Diagnostic:
+                    Console.ForegroundColor = ConsoleColor.DarkGray;
+                    break;
                 case LogLevel.Verbose:
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/ConsoleLogListener.cs]: remove.

**Comment by vwxyzh** [comment]
>         public LogLevel LogLevelThreshold { get; set; }

=> LogLevel.Verbose

---
Refers to: src/Microsoft.DocAsCode.Common/Loggers/ConsoleLogListener.cs:12 in 4f47fa5. [](commit_id = 4f47fa5b1164d83f842920872ffecbbb4d57170d, deletion_comment = False)

**Comment by vwxyzh** [comment]
:clock1:

File: src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ public sealed class ReportLogListener : ILoggerListener
     {
         private readonly StreamWriter _writer;
 
-        public LogLevel LogLevelThreshold { get; set; }
+        public LogLevel LogLevelThreshold => LogLevel.Diagnostic;
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs]: >Diagnostic [](start = 54, length = 10)

can we set this in ctor?

**Comment by herohua** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/Loggers/ILoggerListener.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ namespace Microsoft.DocAsCode.Common
 
     public interface ILoggerListener : IDisposable
     {
-        LogLevel LogLevelThreshold { get; set; }
+        LogLevel LogLevelThreshold { get; }
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Common/Loggers/ILoggerListener.cs]: >LogLevelThreshold [](start = 17, length = 17)

why do we need to expose this? From the code seems it's each listener to decide whether to record the log or not.

**Comment by chenkennt** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**